plugins/woocommerce-admin/client/task-lists/fills/products/utils.tsx

```diff
@@ -2,31 +2,47 @@
  * External dependencies
  */
 import { intersection } from 'lodash';
+import { applyFilters } from '@wordpress/hooks';
 
 /**
  * Internal dependencies
  */
 import {
-	productTypes,
+	productTypes as baseProductTypes,
 	ProductType,
 	ProductTypeKey,
 	onboardingProductTypesToSurfaced,
 	supportedOnboardingProductTypes,
 	defaultSurfacedProductTypes,
+	SETUP_TASKLIST_PRODUCT_TYPES,
 } from './constants';
 
 export const getProductTypes = ( {
 	exclude,
 }: {
 	exclude?: ProductTypeKey[];
 } = {} ): ProductType[] => {
+	/**
+	 * Filter for adding custom product types to tasklist.
+	 *
+	 * @filter woocommerce_tasklist_experimental_product_types
+	 * @param {Object} productTypes Array of product types.
+	 */
+	let productTypes = [
+		...baseProductTypes,
+		...( applyFilters(
```

Comment:
I think we should make the extension point safer. 

AFAIK spreading (`...`) a non-iteratable object may result in an error. Let's check if the filter value is iteratable first.